knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/placeholders/UriInfoHelper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Knot.x - Reactive microservice assembler - Templating Engine Verticle
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.engine.placeholders;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.cognifide.knotx.api.UriInfo;
+
+public final class UriInfoHelper {
+
+    /**
+     * This regular expression is used for decomposite uri into following parts:
+     * <p>
+     * <ul>
+     * <li>path</li>
+     * <li>selector string</li>
+     * <li>extension</li>
+     * <li>suffix</li>
+     * </ul>
+     * <p>
+     */
+    private static final Pattern URI_PATTERN = Pattern.compile(
+            "([^\\.\\?#]+)(\\.([^\\/\\?#]+)\\.)?(\\.?([^\\/\\?#]+))?([^\\?#]+)?((\\?|#).+)?");
```

Comment:
We have enough unit tests for this regular expression. Thank you!

Why aren't we using [JAVA URI](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html) class?

That would also address my other [concerns about URI path](https://github.com/Cognifide/knotx/pull/68/files#r77844369)